daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/processors/unparsers/UState.scala

```diff
@@ -291,6 +291,13 @@ class UStateMain private (
   private var _prior: UStateForSuspension = null
   override def prior = _prior
 
+  override def unparse1(unparser: Unparser): Unit = {
```

Comment:
Hmmm. Doublecheck. because there are multiple kinds of UStates, and only this one does the set/unset of the processor, maybe this wierd call-back structure is needed?

If so, lots of scaladoc explaining why.